lib_xud/src/core/included/XUD_Token_Out_DI.S

```diff
@@ -13,19 +13,24 @@ Pid_Out:
     bf         r3, XUD_TokenOut_BufferFull
     ldw        r1, r3[3]                        // Load buffer from EP structure
 
-CheckEpTypeOut:
-    ldw        r11, r3[5]                       // Load EP type
-    BRFT_ru6   r11, DoOutNonIso                 // ISO endpoint
-
 OutReady:
     BLRF_u10    doRXData                        // Leaves r1: 0
 RxRet2:
-    ldw         r11, r3[1]                      // Load EP chanend
+CheckEpTypeOut:
+    {ldc       r1, 0;                ldw        r11, r3[5]}         // Load EP type
+    bf         r11, StoreTailDataOut            // ISO endpoint
+DoOutNonIso:
```

Comment:
I guess this label should move down a line (though realise it's just informational now)

Reply:
good point